libvast/include/vast/concept/printable/vast/json.hpp

```diff
@@ -85,13 +85,11 @@ struct json_printer
         if (!std::isfinite(x))
           return printers::str.print(out_, "null");
         auto str = std::to_string(x);
-        real i;
         if constexpr (std::is_floating_point_v<T>) {
-          if (std::modf(x, &i) == 0.0)
-            // Do not show 0 as 0.0.
-            str.erase(str.find('.'), std::string::npos);
+          // Avoid trailing zeros.
+          if (real i; std::modf(x, &i) == 0.0)
+            str.erase(str.find('.') + 2, std::string::npos);
```

Comment:
Is it guaranteed that the index is always valid? If not we need to check, since `erase()` will throw `std::out_of_bounds`.

Reply:
Based on https://en.cppreference.com/w/cpp/string/basic_string/to_string it should be.